.circleci/config.yml

```diff
@@ -69,80 +74,50 @@ jobs:
-          command: |
-            git clone git@github.com:fsaintjacques/semver-tool.git
+      - gcp-gcr/gcr-auth
       - restore_cache:
           keys:
-            - v1-{{ .Branch }}
+            - v<< parameters.cache_version>>-cas-shelf-{{ .Branch }}
           paths:
-            - /caches/cas-shelf-tfe-add-app.tar
+            - /tmp/cas-shelf.tar
       - run:
-          name: Load Docker image layer cache
+          name: Load cache
           command: |
             set +o pipefail
-            docker load -i /caches/cas-shelf-tfe-add-app.tar | true
+            docker load -i /tmp/cas-shelf.tar | true
       - run:
           name: Log in to DockerHub
           command: echo $DOCKERHUB_PASSWORD | docker login -u $DOCKERHUB_USER --password-stdin registry.hub.docker.com
       - run:
           name: Build application Docker image
           command: |
-            docker build --cache-from=cas-shelf-tfe-add-app -t cas-shelf-tfe-add-app -f ./docker/add-app/Dockerfile .
+            docker build --cache-from=cas-shelf -t cas-shelf:latest -t gcr.io/${GOOGLE_PROJECT_ID}/cas-shelf:<< parameters.image_tag >> -f ./docker/add-app/Dockerfile .
       - run:
           name: Save Docker image layer cache
           command: |
-            mkdir -p /caches
-            docker save -o /caches/cas-shelf-tfe-add-app.tar cas-shelf-tfe-add-app
+            docker save -o /tmp/cas-shelf.tar cas-shelf:latest
       - save_cache:
-          key: v1-{{ .Branch }}-{{ epoch }}
+          key: v<< parameters.cache_version>>-cas-shelf-{{ .Branch }}-{{ epoch }}
           paths:
-            - /caches/cas-shelf-tfe-add-app.tar
-      - deploy:
-          name: Push application Docker image
+            - /tmp/cas-shelf.tar
+      - gcp-gcr/push-image:
+          image: cas-shelf
+          tag: << parameters.image_tag >>
+          digest-path: /tmp/digest.txt
+      - run:
           command: |
-            SEMVER_GET="./semver-tool/src/semver get"
-            MAJOR=$($SEMVER_GET major $CIRCLE_TAG)
-            MINOR=$($SEMVER_GET minor $CIRCLE_TAG)
-            PATCH=$($SEMVER_GET patch $CIRCLE_TAG)
-            PRE=$($SEMVER_GET prerel $CIRCLE_TAG)
-
-            TAG1=${MAJOR}
-            TAG2=${MAJOR}.${MINOR}
-            TAG3=${MAJOR}.${MINOR}.${PATCH}
-            TAG4=${MAJOR}.${MINOR}.${PATCH}-${PRE}
-
-            ALL_TAGS=""
-            ALL_TAGS="${ALL_TAGS} ${TAG1}"
-            ALL_TAGS="${ALL_TAGS} ${TAG2}"
-            ALL_TAGS="${ALL_TAGS} ${TAG3}"
-            ALL_TAGS="${ALL_TAGS} ${TAG4}"
-
-            if [ $CIRCLE_TAG != $TAG4 ]; then
-              ALL_TAGS="${ALL_TAGS} ${CIRCLE_TAG}"
-            fi
-
-            docker login docker.pkg.github.com -u $GITHUB_USERNAME -p $GITHUB_TOKEN
-
-            for TAG in ${ALL_TAGS}; do
-              TARGET="docker.pkg.github.com/bcgov/cas-shelf/cas-shelf-tfe-add-app:${TAG}"
-              docker tag cas-shelf-tfe-add-app ${TARGET}
-              docker push ${TARGET}
-            done
```

Comment:
We should keep this to add all the appropriate tags to the docker image, as documented in the readme: https://github.com/bcgov/cas-shelf#docker-image-tag-convention